File: minimove/Assets/PSMoveColor.cs
using UnityEngine;
using System.Collections.Generic;

public class PSMoveColor {
	private static readonly System.Random rnd = new System.Random();
	private static Dictionary<PSMoveButton, Color> buttonToColor = new Dictionary<PSMoveButton, Color>() {
		{ PSMoveButton.Square, Color.magenta },
		{ PSMoveButton.Triangle, Color.green },
		{ PSMoveButton.Cross, Color.blue },
		{ PSMoveButton.Circle, Color.red },
	};
	private static List<Color> colors;

	static PSMoveColor() {
		colors = new List<Color>();
		foreach (var color in buttonToColor.Values) {
			colors.Add(color);
		}
	}

	public static Color getRandomColor() {
		return colors[rnd.Next(colors.Count)];
	}

	public static bool isColorButton(PSMoveButton button) {
		return buttonToColor.ContainsKey (button);
	}

	public static Color colorForButton(PSMoveButton button) {
		return buttonToColor [button];
	}
}

File: python/psmovecolor.py
import random
import psmoveapi

from color import Color

class PSMoveColor(object):
    BUTTON_TO_COLOR = {
        psmoveapi.Button.SQUARE: Color.MAGENTA,
        psmoveapi.Button.TRIANGLE: Color.GREEN,
        psmoveapi.Button.CROSS: Color.BLUE,
        psmoveapi.Button.CIRCLE: Color.RED,
    }

    COLORS = list(BUTTON_TO_COLOR.values())

    @classmethod
    def get_random_color(cls):
        return random.choice(cls.COLORS)

    @classmethod
    def is_color_button(cls, button):
        return button in cls.BUTTON_TO_COLOR.keys()

    @classmethod
    def color_for_button(cls, button):
        return cls.BUTTON_TO_COLOR[button]
